hack/push-cip.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About: This script builds and pushes the cip and auditor images to a public repository,
+# defined in workspace_status.sh.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+repo_root=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)
+# Detect which container variant to build.
+build_variant="cip"
+
+printUsage() {
+    >&2 echo "Usage: $0 [--audit]"
+}
+
+if [ $# -gt 1 ] || { [ $# -gt 0 ] && [ "$1" != --audit ];}; then
+    >&2 echo "ERROR: Invalid arguments."
+    printUsage
+    exit 1
+elif [ $# == 1 ] && [ "$1" == --audit ]; then
```

Comment:
Uh, shouldn't this be `(( $# == 1 ))` for arithmetic conditionals?

Also I would use `(( ... ))` so you don't have to use the baroque `-gt` and such flags. Also, prefer `[[` over `[`.

Reply:
Modified all conditions.